client/src/components/Editor/DraftEditor.js

```diff
@@ -0,0 +1,236 @@
+import React, { useState, useEffect, useRef } from "react";
+// Draftjs editor imports
+import Draft, { Editor, EditorState, RichUtils, convertToRaw } from "draft-js";
+import CodeUtils from "draft-js-code";
+import PrismDecorator from "draft-js-prism";
+import Prism from "prismjs";
+import "prismjs/themes/prism-okaidia.css";
+import "./css/draft.css";
+// Material-UI imports
+import { makeStyles } from "@material-ui/core/styles";
+import IconButton from "@material-ui/core/IconButton";
+import Icon from "@material-ui/core/Icon";
+import Grid from "@material-ui/core/Grid";
+import Box from "@material-ui/core/Box";
+// Prism language support:
+import {
+  go,
+  clike,
+  ruby,
+  php,
+  python,
+  c,
+  cpp,
+  csharp,
+  java,
+  languagesGrammar,
+} from "./languages";
+
+const useStyles = makeStyles((theme) => ({
+  editor: {
+    border: "1px solid #ddd",
+    borderRadius: "4px",
+  },
+  controls: {
+    borderBottom: "1px solid #ddd",
+  },
+}));
+
+const MyEditor = ({ language, makeSubmit, onSubmit, hasContent }) => {
+  const classes = useStyles();
+
+  let mappedLanguage;
+  if (language === "") {
+    mappedLanguage = null;
+  } else {
+    mappedLanguage = languagesGrammar[language];
+  }
+
+  const decorator = new PrismDecorator({
+    prism: Prism,
+    defaultSyntax: mappedLanguage,
+  });
+
+  const [editorState, setEditorState] = useState(
+    EditorState.createEmpty(decorator)
+  );
+  //Amending code blocks for Prism
+  const getBlockStyle = (block) => {
+    switch (block.getType()) {
+      case "code-block":
+        return "language-".concat(mappedLanguage);
```

Comment:
does this work if mappedLanguage is null?

Reply:
Yes, works fine.